diffyscan/utils/hardhat.py

```diff
@@ -0,0 +1,78 @@
+import os
+import subprocess
+import signal
+
+from urllib.parse import urlparse
+from .logger import logger
+
+
+class Hardhat:
+    sub_process = None
+    TIMEOUT_FOR_INIT_SEC = 5
+
+    def __init__(self):
+        pass
+
+    def is_port_in_use(self, parsed_url) -> bool:
+        import socket
+
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            return s.connect_ex((parsed_url.hostname, parsed_url.port)) == 0
+
+    def start(self, main_config_relative_path: str, binary_config):
+        parsed_url = urlparse(binary_config["local_RPC_URL"])
```

Comment:
usually it's quite a bad idea to store RPC URL inside the config file because it might contain restricted access tokens that shouldn't be published in the sourcecode repo

the approach should be to load those:
- Etherscan token
- RPC URL(s)
- GitHub token

from private non-disclosed env :warning:

Reply:
done